pallets/sto/src/benchmarking.rs

```diff
@@ -0,0 +1,215 @@
+#![cfg(feature = "runtime-benchmarks")]
+use crate::*;
+use core::convert::TryFrom;
+use frame_benchmarking::benchmarks;
+use frame_support::traits::Get;
+use frame_system::RawOrigin;
+use pallet_settlement::{benchmarking::compliance_setup, VenueDetails};
+use polymesh_common_utilities::benchs::User;
+use polymesh_common_utilities::{asset::AssetType, benchs::UserBuilder};
+use polymesh_primitives::TrustedIssuer;
+
+pub type Asset<T> = pallet_asset::Module<T>;
+pub type ComplianceManager<T> = pallet_compliance_manager::Module<T>;
+pub type Identity<T> = identity::Module<T>;
+pub type Timestamp<T> = pallet_timestamp::Module<T>;
+pub type Settlement<T> = pallet_settlement::Module<T>;
+pub type Sto<T> = crate::Module<T>;
+
+fn create_asset<T: Trait>(
+    origin: RawOrigin<T::AccountId>,
+    ticker: Ticker,
+    supply: u128,
+) -> DispatchResult {
+    <Asset<T>>::create_asset(
+        origin.into(),
+        vec![b'A'].into(),
+        ticker,
+        supply.into(),
+        true,
+        AssetType::default(),
+        vec![],
+        None,
+    )
+}
+
+fn add_compliance<T: Trait>(origin: RawOrigin<T::AccountId>, ticker: Ticker) -> DispatchResult {
+    //TODO: replace with worst case rules
+    <ComplianceManager<T>>::add_compliance_requirement(origin.into(), ticker, vec![], vec![])
+}
+
+fn generate_tiers<T: Trait>(n: u32) -> Vec<PriceTier<T::Balance>> {
+    let n = n as usize;
+    let mut tiers = Vec::with_capacity(n);
+    for i in 0..n {
+        tiers.push(PriceTier {
+            total: 1.into(),
+            price: ((i + 1) as u128).into(),
+        })
+    }
+    tiers
+}
+
+fn setup_fundraiser<T: Trait>() -> Result<(User<T>, Ticker), DispatchError> {
+    let alice = <UserBuilder<T>>::default().generate_did().build("alice");
+    let alice_portfolio = PortfolioId::default_portfolio(alice.did());
+
+    let offering_ticker = Ticker::try_from(&[b'A'][..]).unwrap();
+    let raise_ticker = Ticker::try_from(&[b'B'][..]).unwrap();
+    create_asset::<T>(alice.origin(), offering_ticker, 1_000_000)?;
+    create_asset::<T>(alice.origin(), raise_ticker, 1_000_000)?;
+
+    add_compliance::<T>(alice.origin(), offering_ticker)?;
+    add_compliance::<T>(alice.origin(), raise_ticker)?;
+
+    let venue_id = <Settlement<T>>::venue_counter();
+    <Settlement<T>>::create_venue(
+        alice.origin().into(),
+        VenueDetails::default(),
+        vec![alice.account()],
+        VenueType::Sto,
+    )?;
+
+    <Sto<T>>::create_fundraiser(
+        alice.origin().into(),
+        alice_portfolio,
+        offering_ticker,
+        alice_portfolio,
+        raise_ticker,
+        generate_tiers::<T>(1),
+        venue_id,
+        None,
+        None,
+    )?;
+
+    Ok((alice, offering_ticker))
+}
+
+benchmarks! {
+    _ {}
+
+    create_fundraiser {
+        // Number of tiers
+        let i in 1 .. MAX_TIERS as u32;
+
+        let alice = <UserBuilder<T>>::default().generate_did().build("alice");
```

Comment:
There is a convenience function for this now in `pallets/common/src/benchs/mod.rs`:
```rust
user::<T>("alice", 0)
```